op-chain-ops/foundry/artifactsfs.go

```diff
@@ -74,7 +74,10 @@ func (af *ArtifactsFS) ListContracts(name string) ([]string, error) {
 // ReadArtifact reads a specific JSON contract artifact from the FS.
 // The contract name may be suffixed by a solidity compiler version, e.g. "Owned.0.8.25".
 func (af *ArtifactsFS) ReadArtifact(name string, contract string) (*Artifact, error) {
-	artifactPath := path.Join(name+".sol", contract+".json")
+	if !strings.HasSuffix(name, ".sol") {
+		name += ".sol"
+	}
+	artifactPath := path.Join(name, contract+".json")
```

Comment:
is this sugar required? I think if we find ourselves with a mix of files with/without extensions, it'd be better to just force ourselves to fix it than to attempt to correct it live.

This also has the interesting effect that if you take the actual `artifactPath` and attempt to use it to open the file, you'll end up with `myFile.sol.json.sol.json`, which is a bit of a foot-gun.

Reply:
Good call, I had some doubts about it too. I changed it to not include the suffix sugar, and updated the related FS methods and doc comments to be explicit about when the file-extension is expected.